### ISSUE TITLE
vobject claims UID MUST be present, does not automatically fix up missing UID

### ISSUE THREAD

**Issue by dagbdagb** [fruux/sabre-vobject#176]
I note that vobject loudly complains if a vcard misses a UID:

vCard: 3.0
  [ERROR] UID MUST appear exactly once in a VCARD component

According to RFC 6350, UID has    Cardinality:  *1

...which means:
    |      *1     | Exactly one instance per vCard MAY be present.   |

Now, RFC 6350 covers vcard 4.0. And also claims to replace RFC 2426 (vcard 3.0).
RFC 2426 only lists FN, N and VERSION as mandatory. 
(side note: N is mandatory in vcard 3.0, but not 4.0).

In short:
I believe vobject is in error claiming a UID MUST be present per vcard.

However, *if* the project manager still thinks this is important to keep, how about autogenerating a UID for those vcards missing one?

**Comment by staabm** [comment]
This was already added recently https://github.com/fruux/sabre-vobject/pull/174

**Comment by dagbdagb** [comment]
Ah. Sorry for the duplicate. Feel free to tag this as such.
Still not convinced UID is mandatory in any version of vcard.

**Comment by evert** [comment]
Going to treat this as a bug. UID is not required for vCards, but for CardDAV it is.

The validator should have a new option that allows people to select a 'CardDAV profile' for vCard validation.

In the case of non-carddav, we can simply reduce the validation error to a warning. It's still a good idea to have a UID.
